pkg/ccl/sqlproxyccl/connector.go

```diff
@@ -319,7 +319,11 @@ func (c *connector) lookupAddr(ctx context.Context) (string, error) {
 	if c.Directory != nil {
 		addr, err := c.Directory.EnsureTenantAddr(ctx, c.TenantID, c.ClusterName)
 		if err != nil {
-			if status.Code(err) != codes.NotFound {
+			if status.Code(err) == codes.FailedPrecondition {
```

Comment:
nit: it would be nice to flatten this logic. I'm thinking something like

```
switch {
case err == nil:
  return add, nil
case status.Code(err) == codes.FailedPrecondition:
  message = "unavailable"
  if st, ok := status.FromError(err); ok {
    message = st.Message()
  }
  return "", newErrorf(codeUnavailable, st.Message()
case status.Code(err) != codes.NotFound:
  return "", markAsRetriableConnectionError(err)
default: 
  /* fallback to old resolution rule */
}
```